identity/app/idapiclient/IdApiClient.scala

```diff
@@ -57,6 +57,14 @@ class IdApiClient(
     response map extractUser
   }
 
+  def userFromQueryParam(param: String, field: String,auth: Auth = Anonymous): Future[Response[User]] = {
```

Comment:
Can `userFromVanityUrl` now call this to consolidate this kind of query logic?

Reply:
It is a slightly different implementation so probably doesn't make sense to try and make one function.